components/fs-data-provider/src/fs_data_provider.rs

```diff
@@ -0,0 +1,73 @@
+use crate::error::Error;
+use crate::manifest::Manifest;
+use icu_data_provider::prelude::*;
+use icu_data_provider::structs;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::PathBuf;
+
+/// A data provider that reads ICU4X data from a filesystem directory.
+#[derive(Debug, PartialEq)]
+pub struct FsDataProvider {
+    root: PathBuf,
+    manifest: Manifest,
+}
+
+impl FsDataProvider {
+    /// Create a new FsDataProvider given a filesystem directory.
+    pub fn try_new<T: Into<PathBuf>>(root: T) -> Result<Self, Error> {
+        let root_path_buf: PathBuf = root.into();
+        let manifest_path = root_path_buf.join("manifest.json");
```

Comment:
This is a good idea, because there are two different places where I use "manifest.json", and you found one but not the other.  I moved the definition into manifest.rs as a constant.